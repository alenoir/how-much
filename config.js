import React from 'react';
import ReactNative from 'react-native';
const Immutable = require('immutable');

const window = global || window;

window.React = React;
window.ReactNative = ReactNative;
window.Immutable = Immutable;
