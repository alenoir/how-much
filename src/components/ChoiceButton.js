const {
  StyleSheet,
  View,
  Dimensions,
  Text,
  TouchableWithoutFeedback,
} = ReactNative;

const {
  Component,
  PropTypes,
} = React;

const { width } = Dimensions.get('window');

const styles = StyleSheet.create({
  content: {
    alignSelf: 'center',
    width: width / 5,
    height: 30,
    alignItems: 'center',
    justifyContent: 'center',
  },
  text: {
    fontFamily: 'Varela Round',
    color: 'white',
    fontSize: 18,
  },
  selected: {
    backgroundColor: '#32BF69',
  },
});

class ChoiceButton extends Component {

  render() {
    const { name, selected, onPress } = this.props;

    const btnStyle = [];
    btnStyle.push(styles.content);
    if (selected) {
      btnStyle.push(styles.selected);
    }

    return (
      <TouchableWithoutFeedback
        key={name}
        onPress={(() => onPress(name))}
      >
        <View style={btnStyle}>
          <Text style={styles.text}>{name}</Text>
        </View>
      </TouchableWithoutFeedback>
    );
  }
}

ChoiceButton.propTypes = {
  selected: PropTypes.bool.isRequired,
  name: PropTypes.string.isRequired,
  onPress: PropTypes.func,
};

module.exports = ChoiceButton;
