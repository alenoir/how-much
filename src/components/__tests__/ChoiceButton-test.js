import renderer from 'react-test-renderer'; // eslint-disable-line

import '../../../config.js';
import ChoiceButton from '../ChoiceButton';

describe('Choice Button', () => {
  it('Render unselected', () => {
    const props = {
      name: 'TestString',
      onPress: () => {},
      selected: false,
    };
    const tree = renderer.create(
      <ChoiceButton {...props} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Render selected', () => {
    const props = {
      name: 'TestString',
      onPress: () => {},
      selected: true,
    };

    const tree = renderer.create(
      <ChoiceButton {...props} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
