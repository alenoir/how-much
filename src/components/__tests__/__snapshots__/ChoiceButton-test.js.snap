exports[`Choice Button Render selected 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function bound touchableHandleResponderGrant]}
  onResponderMove={[Function bound touchableHandleResponderMove]}
  onResponderRelease={[Function bound touchableHandleResponderRelease]}
  onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
  onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
  onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "height": 30,
        "justifyContent": "center",
        "width": 75
      },
      Object {
        "backgroundColor": "#32BF69"
      }
    ]
  }
  testID={undefined}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontFamily": "Varela Round",
        "fontSize": 18
      }
    }>
    TestString
  </Text>
</View>
`;

exports[`Choice Button Render unselected 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function bound touchableHandleResponderGrant]}
  onResponderMove={[Function bound touchableHandleResponderMove]}
  onResponderRelease={[Function bound touchableHandleResponderRelease]}
  onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
  onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
  onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "height": 30,
        "justifyContent": "center",
        "width": 75
      }
    ]
  }
  testID={undefined}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontFamily": "Varela Round",
        "fontSize": 18
      }
    }>
    TestString
  </Text>
</View>
`;
